### PR TITLE
QueryResult and QueryBatchCursor renaming

### PR DESCRIPTION
**Pull request by rozza**
Renamed and moved QueryResult to CommandCursorResult Renamed QueryBatchCursor to CommandBatchCursor
Renamed AsyncQueryBatchCursor to AsyncCommandBatchCursor

Part of a wider Command Cursor refactoring

JAVA-5159

### CODE REVIEW

File: driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java [hunk 1]
```diff
@@ -84,11 +94,21 @@ public List<T> getResults() {
...
      */
-    public ServerAddress getAddress() {
+    public ServerAddress getServerAddress() {
         return serverAddress;
     }
 
     public long getCursorId() {
         return cursorId;
     }
+
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java:104]: These methods are unused, even by tests?  Maybe they are used in a subsequent PR?

Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java:104]: Correct - these PRs are just extra reworkings of the original PR.

**Review by jyemin — APPROVED** [review]

File: driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java [hunk 2]
```diff
@@ -14,48 +14,58 @@
...
  * <p>This class is not part of the public API and may be removed or changed at any time</p>
  */
-public class QueryResult<T> {
-    private final MongoNamespace namespace;
+public class CommandCursorResult<T> {
+
+    private static final String CURSOR = "cursor";
+    private static final String POST_BATCH_RESUME_TOKEN = "postBatchResumeToken";
+    private static final String OPERATION_TIME = "operationTime";
+    private final BsonDocument cursorDocument;
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java:40]: Intellij points out that this field can be converted to a local variable in the constructor.

**Review by jyemin — CHANGES_REQUESTED** [review]

File: driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java (hunk 2 above)
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/CommandCursorResult.java]: Done

**Review by jyemin — APPROVED** [review]
LGTM

**Comment by rozza** [comment]
Closed and rolled up to #401

**Pull request closed without merging**